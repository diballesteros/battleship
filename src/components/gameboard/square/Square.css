.square {
    width: 50px;
    height: 50px;
    border: 1px solid;
    font-size: 32px;
    text-align: center;
    cursor: pointer;
}
.my_square {
    cursor: default;
}

.ocean-square {
    background-color: skyblue;
}
.ship-square {
    background-color: grey;
}

