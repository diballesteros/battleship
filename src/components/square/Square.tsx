import React, { memo } from 'react';
import styles from './Square.module.scss';

const onDragOver = (e: React.DragEvent<HTMLDivElement>) => {
  // e.target.style.backgroundColor = 'darkblue';
  e.preventDefault();
};

const onDragLeave = (e: React.DragEvent<HTMLDivElement>) => {
  // e.target.style.removeProperty('background-color');
  e.preventDefault();
};

interface SquareProps {
  height?: number | string;
  width?: number | string;
}

const Square: React.FC<SquareProps> = memo(({ height, width }) => {
  return (
    <div
      className={`${styles.square} ${styles.ocean}`}
      style={{
        height: height || undefined,
        minHeight: height || undefined,
        width: width || undefined,
        maxWidth: width || undefined,
      }}
      onDrop={null}
      onDragOver={onDragOver}
      onDragLeave={onDragLeave}
      role="presentation"
    />
  );
});

export default Square;
