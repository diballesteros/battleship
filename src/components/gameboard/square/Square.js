import React from 'react';
import './Square.css';

const Square = (props) => {
    switch (props.type) {
        case "ship":
            return <div onClick={props.receiveAttack} className={`square ship-square ${props.myBoard ? 'my_square' : ''}`} data-testid={props.id}>{props.hit ? 'X' : ''}</div>
        default:
            return <div onClick={props.receiveAttack} className={`square ocean-square ${props.myBoard ? 'my_square' : ''}`}>{props.hit ? 'X' : ''}</div>
    }
}

export default Square;
