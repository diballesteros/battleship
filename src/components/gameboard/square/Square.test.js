import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Square from './Square';

describe('Square component', () => {
    test('Ocean Square initialized', () => {
        const { asFragment } = render(<Square type='ocean' hit={false} />);
        expect(asFragment()).toMatchSnapshot();
    });

    test('Ship Square initialized', () => {
        const { asFragment } = render(<Square type='ship' hit={false} />);
        expect(asFragment()).toMatchSnapshot();
    });

    test('Ship square hit', () => {
        const mockFn = jest.fn();
        const { container } = render(<Square type='ship' hit={false} hitHandler={() => mockFn()} />);
        const component = container.querySelector('div');
        fireEvent.click(component);
        expect(mockFn).toHaveBeenCalled();
    });
});