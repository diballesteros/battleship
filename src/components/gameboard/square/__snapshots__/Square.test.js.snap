// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Square component Ocean Square initialized 1`] = `
<DocumentFragment>
  <div
    class="square ocean-square "
  />
</DocumentFragment>
`;

exports[`Square component Ship Square initialized 1`] = `
<DocumentFragment>
  <div
    class="square ship-square "
  />
</DocumentFragment>
`;
